experimenter/experimenter/segments/__init__.py

```diff
@@ -68,22 +68,22 @@ def clear_cache(cls):
         cls._segments = None
 
     @classmethod
-    def all(cls):
+    def all(cls, segment_data=None):
```

Comment:
I'd suggest adding type hints to segment_data since it may not be immediately obvious what it is intended to be (here and in `by_application`).